interpreter/src/test/java/tw/waterball/survive/io/interpreter/impl/StreamLiteralReaderTest.java

```diff
@@ -0,0 +1,105 @@
+package tw.waterball.survive.io.interpreter.impl;
+
+import org.junit.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.EOFException;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+import static org.junit.Assert.*;
+
+/**
+ * TODO:
+ *  - column / row / position calculation isn't strictly tested and implemented
+ * @author Waterball (johnny850807@gmail.com)
+ */
+public class StreamLiteralReaderTest {
+    private StreamLiteralReader literalReader;
+
+    @Test
+    public void givenText_Apple_whenNextChar3Times_shouldBe_A_p_p_respectively() throws IOException {
+        givenText("Apple");
+
+        assertSame('A', literalReader.nextChar());
+        assertSame('p', literalReader.nextChar());
+        assertSame('p', literalReader.nextChar());
+    }
+
+    @Test
+    public void givenText_Apple_goToPos3_whenGetCurrentChar_shouldBe_p() throws IOException {
+        givenText("Apple");
+        goToPosition(3);
```

Comment:
Would the users of this API expect that it should start from 0? But I know you mean when print out the error message it should start from 1.

Reply:
Thanks for the comment, actually it starts from 0 exactly.

Reply:
The API may be improved in future adjustments if any confusion intensified.